metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1406,44 +1406,51 @@ class MetalsLanguageServer(
         .orNull
     }.asJava
 
-  private def supportedBuildTool(): Option[BuildTool] =
+  private def supportedBuildTool(): Future[Option[BuildTool]] =
     buildTools.loadSupported match {
-      case Some(buildTool) =>
+      case Nil => {
+        if (!buildTools.isAutoConnectable) {
+          warnings.noBuildTool()
+        }
+        Future(None)
+      }
+      case buildTool :: Nil => {
         val isCompatibleVersion = SemVer.isCompatibleVersion(
           buildTool.minimumVersion,
           buildTool.version
         )
         if (isCompatibleVersion) {
-          Some(buildTool)
+          Future(Some(buildTool))
         } else {
           scribe.warn(s"Unsupported $buildTool version ${buildTool.version}")
           languageClient.showMessage(
             Messages.IncompatibleBuildToolVersion.params(buildTool)
           )
-          None
-        }
-      case None =>
-        if (!buildTools.isAutoConnectable) {
-          warnings.noBuildTool()
+          Future(None)
         }
-        None
+      }
+      case buildTools =>
+        bloopInstall.checkForChosenBuildTool(buildTools)
```

Comment:
Can we extract `isCompatibleVersion` to a function (maybe just inner function) and use it both for 2. and 3. case ? This way we will not need to do it in `checkForChosenBuildTool`

Reply:
Yup, done.